app/models/mobile_user.rb

```diff
@@ -5,8 +5,14 @@ class MobileUser < ApplicationRecord
   validates :auth_token, presence: true
   validates :name, presence: true, unless: :edge?
 
-  def image_url
-    AvatarGenerator.new(auth_token.value).generate_url
+  before_save :set_color
+
+  def set_color
+    self.avatar_color = InitialsAvatarGenerator.call
+  end
+
+  def initials
+    [self.first_name, self.last_name].compact.map { |s| s.first }.join(' ')
```

Comment:
Zamiast `map { |s| s.first }` można pisać `map(&:first)` - jest mniej pisania, poza tym jest to szybsze (jakieś mikrosekundy, więc mało ważne, ale zawsze).